packages/fast-components-react-base/src/horizontal-overflow/horizontal-overflow.tsx

```diff
@@ -76,6 +90,43 @@ class HorizontalOverflow extends Foundation<HorizontalOverflowProps,  React.AllH
         this.setState({
             itemsHeight
         });
+
+        if (canUseDOM() && this.horizontalOverflowItemsRef.current) {
+            this.horizontalOverflowItemsRef.current.addEventListener("scroll", this.throttled);
+        }
+    }
+
+    /**
+     * React life-cycle method
+     */
+    public componentWillUnmount(): void {
+        if (canUseDOM() && this.horizontalOverflowItemsRef.current) {
+            this.horizontalOverflowItemsRef.current.removeEventListener("scroll", this.throttled);
+        }
+    }
+
+    /**
+     * Callback on scroll change
+     */
+    public onScrollChange = (): IScrollChange => {
+        let scrollObject: IScrollChange;
+        const isLtr: boolean = this.getLTR() === Direction.ltr;
+        const distanceRemaining: number =
+            this.horizontalOverflowItemsRef.current.scrollWidth - this.horizontalOverflowItemsRef.current.scrollLeft;
+
+        if (this.horizontalOverflowItemsRef.current.scrollLeft === 0) {
+            scrollObject = { start: isLtr, end: !isLtr };
+        } else if (distanceRemaining === this.horizontalOverflowItemsRef.current.clientWidth) {
+            scrollObject = { start: !isLtr, end: isLtr };
+        } else {
+            scrollObject = { start: false, end: false };
+        }
+
+        if (this.props.onScrollChange) {
```

Comment:
This should be a check at the beginning, if the onScrollChange is not present just return.

Reply:
How would we pass them the scroll object with data if it is at the beginning?

Reply:
Her point is that this shouldn't return anything - consumers get notified via a registered callback. Because we only care about this data if a callback is registered, we can check for the existence of the callback before doing any work. If the callback doesn't exist, we know we don't need to do any work and we can just exit because the point of this function is to call a consumer-registered callback.